Content.Shared/Interaction/SharedInteractionSystem.cs

```diff
@@ -433,55 +366,92 @@ protected bool ValidateInteractAndFace(EntityUid user, EntityCoordinates coordin
             float range = InteractionRange,
             CollisionGroup collisionMask = CollisionGroup.Impassable,
             Ignored? predicate = null,
-            bool ignoreInsideBlocker = false,
             bool popup = false)
         {
-            predicate ??= e => e == origin || e == other;
-            return InRangeUnobstructed(origin, Transform(other).MapPosition, range, collisionMask, predicate, ignoreInsideBlocker, popup);
+            var originPosition = Transform(origin).MapPosition;
+            var transform = Transform(other);
+            var (position, rotation) = transform.GetWorldPositionRotation();
+            var mapPos = new MapCoordinates(position, transform.MapID);
+            var wallPredicate = AddAnchoredPredicate(other, mapPos, rotation, originPosition);
+
+            Ignored combinedPredicate = e =>
+            {
+                return e == origin
+                    || e == other
+                    || (predicate?.Invoke(e) ?? false)
+                    || (wallPredicate?.Invoke(e) ?? false);
+            };
+
+            var inRange = InRangeUnobstructed(origin, mapPos, range, collisionMask, combinedPredicate, popup);
+
+            if (!inRange && popup)
+            {
+                var message = Loc.GetString("interaction-system-user-interaction-cannot-reach");
+                origin.PopupMessage(message);
```

Comment:
Does popupsystem have a method for this?

Reply:
yeah it does. This is the problem with proxy methods. So convenient yet they hide all the resolves & gets.